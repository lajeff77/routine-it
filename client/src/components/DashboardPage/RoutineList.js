import React, {useState} from 'react';
import axios from 'axios';
import {useNavigate} from 'react-router-dom';
import {Box, Button,Divider, List, Typography} from '@mui/material';
import Routine from './Routine';

const RoutineList = (props) => {
    const [tasks, setTasks] = useState([]);
    const navigate = useNavigate();

    const newRoutine = () =>{
        navigate("/newroutine");
    }

    const getTasks = async (routineId) => {
        try{
            const response = await axios.get(`http://localhost:8080/api/v1/tasks/${routineId}`);
            setTasks(response.data)
        
        }catch(error){
            console.log(`an error occurred while fetching the tasks to calculate the percentage: ${error}`)
        }
    }

    const calculateRoutineProgress = (routineId, mode) =>{
        //parse routine with level to see how many 
        // TODO call tasks api here
        getTasks(routineId);
        //sort tasks
        let totalTasks = 0;
        let completedTasks = 0;
        for(let i = 0; i < tasks.length; i++)
        {
            if(tasks[i].highEffort && mode === "HIGH")
            {
                totalTasks++;
                if(tasks[i].complete)
                    completedTasks++;
            }
                
            if(tasks[i].mediumEffort && mode === "MEDIUM")
            {
                totalTasks++;
                if(tasks[i].complete)
                    completedTasks++;
            }
                
            if(tasks[i].lowEffort && mode === "LOW")
            {
                totalTasks++;
                if(tasks[i].complete)
                    completedTasks++;
            }
                
        }
        let percent = (totalTasks/completedTasks) *100;
        console.log(`percent is ${percent}`);
        return percent.toString();
    }

    return(
        <Box sx={{margin:'100px'}}>
            <Typography variant="h3">Active Routines</Typography>
            <List>
            {props.routineList.map((routine) => (
                        <Box>
                            <Routine name={routine.name} routine={routine} id={routine.id} progress={() => calculateRoutineProgress(routine.id, routine.mode)} mode={routine.mode}/>
                            <Divider/>
                        </Box>
                    ))}
            </List>
            <Button variant="contained" sx={{width:"100%", margin:"10px"}} onClick={newRoutine}>Create New Routine</Button>
        </Box>
    );
};

export default RoutineList;