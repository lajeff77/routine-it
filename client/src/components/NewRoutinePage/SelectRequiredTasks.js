import React, {useEffect, useState} from 'react';
import {Box, Card, Typography, Checkbox, FormGroup, FormControlLabel} from '@mui/material';
import {Formik, Form} from 'formik';

const SelectRequiredTasks = ({props}) => {
    const [checkedItems, setCheckedItems] = useState([]);

    let formik = props.formik;

    useEffect(() => {
        formik.validateForm();
    },[])

    useEffect(() => {
        switch(props.id){
            case 0:
                formik.setFieldValue("lowEffortRoutine", checkedItems);
                break;
            case 1:
                formik.setFieldValue("medEffortRoutine", checkedItems);                
                break;
            case 2:
                formik.setFieldValue("highEffortRoutine", checkedItems);
                break;
            default:
                //do nothing
                break;
        }
    }, [checkedItems]);

    const handleChange = (event) => {
        const { name, checked } = event.target;
        if (checked) {
            setCheckedItems(prevItems => [...prevItems, name]);
        } else {
            setCheckedItems(prevItems => prevItems.filter(item => item !== name));
        }
    };

    return(
        <Box sx={{margin:"20px", display:"flex", flexDirection:"column", alignItems:"center"}}>
            <Typography variant="h1" sx={{textAlign: 'center', margin: '30px'}}>{props.text.title}</Typography>
            <Card variant="outlined" sx={{align:"center", width:"70%", display:"flex", flexDirection:"column", outline:"2px"}}>
                <Typography variant="body1"  sx={{margin: '20px'}}>{props.text.prompt}</Typography>
                <Box>
                    <Formik initialValues={{list: checkedItems}} validateOnChange={true}>
                        <Form name="list">
                            {formik.values.tasks.map((item)=> {return(
                                <Box>
                                    <FormGroup sx={{margin:"20px"}}>
                                        <FormControlLabel control={<Checkbox name={item} checked={checkedItems.includes(item)} onChange={handleChange}/>} label={item} />
                                    </FormGroup>
                                </Box>)
                            })}
                        </Form>
                    </Formik> 
                 </Box>   
            </Card>
        </Box>
    )
}

export default SelectRequiredTasks;

