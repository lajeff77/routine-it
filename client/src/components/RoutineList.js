import React from 'react';
import {useNavigate} from 'react-router-dom';
import {Box, Button,Divider, List, Typography} from '@mui/material';
import Routine from './Routine';

const RoutineList = () => {
    const navigate = useNavigate();

    const newRoutine = () =>{
        navigate("/newroutine");
    }

    return(
        <Box sx={{margin:'100px'}}>
            <Typography variant="h3">Active Routines</Typography>
            <List>
                <Routine name="Morning Routine" progress="100"/>
                <Divider/>
                <Routine name="Exercise Routine" progress="95"/>
                <Divider/>
                <Routine name="Study Routine" progress="30"/>
                <Divider/>
                <Routine name="Night Routine" progress="0"/>
            </List>
            <Button variant="contained" sx={{width:"100%", margin:"10px"}} onClick={newRoutine}>Create New Routine</Button>
        </Box>
    );
};

export default RoutineList;