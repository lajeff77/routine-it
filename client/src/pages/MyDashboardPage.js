import React from "react";
import Dashboard from "../components/Dashboard";
import PermanentSideDrawer from "../components/PermanentSideDrawer";
import {Grid, Item} from '@mui/material';

const MyDashboardPage = () => {

    return (
        <div>
            <Grid container rowSpacing={2} >
                <Grid item xs={2}><PermanentSideDrawer /></Grid>
                <Grid item xs={10}><Dashboard /></Grid>
            </Grid>
        </div>
    )
}

export default MyDashboardPage;