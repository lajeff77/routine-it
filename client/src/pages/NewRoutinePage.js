import React, {useState} from 'react';
import {Box} from '@mui/material';
import NameRoutine from '../components/NameRoutine';
import SurveyNav from '../components/SurveyNav';

const NewRoutinePage = () => {
    const [index,setIndex] = useState(0);
    const survey =[<NameRoutine/>];

    return(
        <Box>
            {survey[index]}
            <SurveyNav props={{setIndex:setIndex, index:index, maxIndex:survey.length-1}}/>
        </Box>
    );
}

export default NewRoutinePage;