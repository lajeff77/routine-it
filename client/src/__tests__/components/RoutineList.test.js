import { render } from "@testing-library/react";
import {expect, test} from '@jest/globals';
import React from "react";
import RoutineList from "../../components/RoutineList";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import MyDashboardPage from "../../pages/MyDashboardPage";
import NewRoutinePage from "../../pages/NewRoutinePage";


describe(RoutineList.name, ()=>{
    test('RoutineList matches snapshot', () => {
        const { asFragment } = render(
            <MemoryRouter>
            <Routes>
                <Route element={<MyDashboardPage />} path="/mydashboard" />
                <Route element={<NewRoutinePage/>} path="/newroutine" />
            </Routes>
            </MemoryRouter>
        );
    
        expect(asFragment()).toMatchSnapshot();
    });
});