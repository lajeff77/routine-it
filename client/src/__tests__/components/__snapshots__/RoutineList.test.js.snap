// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutineList RoutineList matches snapshot 1`] = `<DocumentFragment />`;
